robot/client/client.go

```diff
@@ -71,17 +72,87 @@ type RobotClient struct {
 	closeContext context.Context
 }
 
+var exemptFromConnectionCheck = map[string]bool{
+	"/proto.rpc.webrtc.v1.SignalingService/Call":                 true,
+	"/proto.rpc.webrtc.v1.SignalingService/CallUpdate":           true,
+	"/proto.rpc.webrtc.v1.SignalingService/OptionalWebRTCConfig": true,
+	"/proto.rpc.v1.AuthService/Authenticate":                     true,
+}
+
+func needsConnectionCheck(method string) bool {
+	return !exemptFromConnectionCheck[method]
+}
+
+const readWriteOnClosedPipeErrorMsg = "read/write on closed pipe"
+
+func isRWOnClosedPipeError(err error) bool {
+	if err == nil {
+		return false
+	}
+	return strings.Contains(err.Error(), readWriteOnClosedPipeErrorMsg)
+}
+
+func (rc *RobotClient) notConnectedToRemoteError() error {
+	return errors.Errorf("not connected to remote robot at %s", rc.address)
+}
+
+func (rc *RobotClient) handleUnaryDisconnect(
+	ctx context.Context,
+	method string,
+	req, reply interface{},
+	cc *googlegrpc.ClientConn,
+	invoker googlegrpc.UnaryInvoker,
+	opts ...googlegrpc.CallOption,
+) error {
+	if needsConnectionCheck(method) {
+		rc.Logger().Debugw("checking remote connection", "method", method)
```

Comment:
this is gonna spam us a lot; maybe remove after testing

Reply:
agreed - I am only going to keep the log for when we skip the method call